prefect_azure/container_instance.py

```diff
@@ -299,6 +308,55 @@ async def run(
             identifier=created_container_group.name, status_code=status_code
         )
 
+    async def kill(self, container_group_name: str, grace_seconds: int = 30):
+        """
+        Kill a flow running in an ACI container group.
+        Args:
+            container_group_name: The container group name yielded by
+                `AzureContainerInstanceJob.run`.
+        """
+        # ACI does not provide a way to specify grace period, but it gives
+        # applications ~30 seconds to gracefully terminate before killing
+        # a container group.
+        if grace_seconds != 30:
+            self.logger.warning(
+                f"Kill grace period of {grace_seconds}s requested, but ACI does not "
+                "support grace period configuration."
+            )
+
+        aci_client = self.aci_credentials.get_container_client(
+            self.subscription_id.get_secret_value()
+        )
+
+        # get the container group to check that it still exists
+        try:
+            container_group = aci_client.container_groups.get(
+                resource_group_name=self.resource_group_name,
+                container_group_name=container_group_name,
+            )
+        except ResourceNotFoundError:
+            # the container group no longer exists, so there's nothing to cancel
+            raise InfrastructureNotFound(
+                f"Cannot stop ACI job: container group {container_group_name} "
+                "no longer exists."
+            )
```

Comment:
Maybe this to link the tracebacks?

```suggestion
        except ResourceNotFoundError as exc:
            # the container group no longer exists, so there's nothing to cancel
            raise InfrastructureNotFound(
                f"Cannot stop ACI job: container group {container_group_name} "
                "no longer exists."
            ) from exc
```